### PR TITLE
Add support for various node roles for ElasticStack

### PR DESCRIPTION
**Pull request by kamolhasan**
Signed-off-by: kamolhasan <kamol@appscode.com>

### CODE REVIEW

File: pkg/distribution/elastic_stack/config.go [hunk 1]
```diff
@@ -82,8 +87,19 @@ func (es *Elasticsearch) EnsureDefaultConfig() error {
...
 	var config string
 
+	// For Elasticsearch version >= 7.9.x
+	// The legacy node role setting (ie. node.master=true) is deprecated,
+	// So, for newer versions, node roles will be managed by elasticsearch.yml file.
+	dbVersion, err := semver.Parse(es.esVersion.Spec.Version)
+	if err != nil {
+		return err
+	}
+	if dbVersion.Major >= 7 && dbVersion.Minor >= 9 {
```
Comment (faem) [inline pkg/distribution/elastic_stack/config.go:97]: It should match for `8.0.x`, `8.1.x` etc too, right? Even if those versions haven't released yet.

File: pkg/controller/reconciler.go [hunk 2]
```diff
@@ -106,10 +106,42 @@ func (r *Reconciler) ReconcileNodes(db *api.Elasticsearch) (*api.Elasticsearch,
...
+		vt10, err := elastic.EnsureTransformNode()
+		if err != nil {
+			return nil, kutil.VerbUnchanged, err
+		}
 
-		if vt1 == kutil.VerbCreated && vt2 == kutil.VerbCreated && vt3 == kutil.VerbCreated {
+		if vt1 == kutil.VerbCreated && vt2 == kutil.VerbCreated && vt3 == kutil.VerbCreated && vt4 == kutil.VerbCreated &&
+			vt5 == kutil.VerbCreated && vt6 == kutil.VerbCreated && vt7 == kutil.VerbCreated && vt8 == kutil.VerbCreated &&
+			vt9 == kutil.VerbCreated && vt10 == kutil.VerbCreated {
 			vt = kutil.VerbCreated
```
Comment (faem) [inline pkg/controller/reconciler.go:141]: If user doesn't provide all kinds of nodes together, this condition will never be true.

**Review by faem — APPROVED** [review]

File: pkg/controller/reconciler.go (hunk 2 above)
Comment (kamolhasan) [reply pkg/controller/reconciler.go]: Solved!

File: pkg/distribution/elastic_stack/config.go (hunk 1 above)
Comment (kamolhasan) [reply pkg/distribution/elastic_stack/config.go]: Updated!

**Review by faem — DISMISSED** [review]
LGTM

**Review by faem — APPROVED** [review]

**Pull request merged**